### DISCUSSION TITLE
Any reason that file watchers are created for each subdirectories instead of just the root folder?

### DISCUSSION THREAD

**Discussion by musjj** [nvim-tree/nvim-tree.lua#1532] [General]
Why not just watch the root directory instead? I suspect that this might have significant performance impact on a root directory with a large amount of subdirectories.

**Comment by alex-courtis** [comment]
libuv watchers are not recursive, as that is not implemented on windows or OSX: http://docs.libuv.org/en/v1.x/fs_event.html

libuv appears to be implemented via inotify (or equivalent) which is very efficient.

git status is only updated for the directory that has changed, not the entire tree. Git status is orders of magnitude more expensive than directory enumeration.

**Reply by musjj** [reply]
>as that is not implemented on windows or OSX:

Isn't it the opposite? From the linked docs:

>Note Currently the only supported flag is UV_FS_EVENT_RECURSIVE and only on OSX and Windows.

**Reply by alex-courtis** [reply]
> Isn't it the opposite? From the linked docs:

Whoops... you're right.

RE performance, it would be a balance between the efficiency of one watcher and all the files under root.

When the flag becomes available we can revisit.
